### PR TITLE
[kong] helm chart tests

### PR DESCRIPTION
**Pull request by shaneutt**
#### What this PR does / why we need it:

This adds [Chart Tests](https://helm.sh/docs/topics/chart_tests/) which can be used to validate the functionality of our Kong chart, and adds CI to run the tests on PRs and merges.

#### Which issue this PR fixes

Partially Resolves https://github.com/Kong/charts/issues/391

#### Special notes for your reviewer:

This is a minimal test set to get us started, I intend to have follow up PRs which add more tests and specifically test upgrades between chart versions.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] PR is based off the current tip of the `next` branch and targets `next`, not `main`
- [x] New or modified sections of values.yaml are documented in the README.md
- [x] Title of the PR and commit headers start with chart name (e.g. `[kong]`)

### CODE REVIEW

**Comment by shaneutt** [comment]
Looks like CI is happy:

```
INFO: running helm tests for all charts on Kubernetes v1.21.1
NAME: chart-tests
LAST DEPLOYED: Wed Jul 14 14:31:19 2021
NAMESPACE: 1d3f5cc3-4cc3-4217-a349-c5b98eddc859
STATUS: deployed
REVISION: 1
TEST SUITE:     chart-tests-test-ingress
Last Started:   Wed Jul 14 14:31:20 2021
Last Completed: Wed Jul 14 14:31:52 2021
Phase:          Succeeded
TEST SUITE:     chart-tests-test-ingress-v1beta1
Last Started:   Wed Jul 14 14:31:52 2021
Last Completed: Wed Jul 14 14:31:54 2021
Phase:          Succeeded
```

Thanks again for helping me with that issue in the CI @rainest :bow:

**Comment by shaneutt** [comment]
At this point these tests add something  meaningful to the repo and they work, while technically I've got more tests to add I would like to merge this as is and spread the PRs out into smaller PRs as I add more tests, so I'm going to mark this as "Ready To Review" but just note you'll be seeing more of these soon.

**Comment by shaneutt** [comment]
One sec, just realized that I'm missing changes from my stash, WIP for a moment.

**Comment by shaneutt** [comment]
Still happy after moving the tasks around in the Github Workflows:

```
INFO: running helm tests for all charts on Kubernetes v1.21.1
Pod chart-tests-test-ingress pending
Pod chart-tests-test-ingress pending
Pod chart-tests-test-ingress pending
Pod chart-tests-test-ingress running
Pod chart-tests-test-ingress running
Pod chart-tests-test-ingress running
Pod chart-tests-test-ingress succeeded
Pod chart-tests-test-ingress-v1beta1 pending
Pod chart-tests-test-ingress-v1beta1 pending
Pod chart-tests-test-ingress-v1beta1 pending
Pod chart-tests-test-ingress-v1beta1 succeeded
NAME: chart-tests
LAST DEPLOYED: Wed Jul 14 14:51:41 2021
NAMESPACE: 70904e91-721c-41a1-9197-b45b64cee49e
STATUS: deployed
REVISION: 1
TEST SUITE:     chart-tests-test-ingress
Last Started:   Wed Jul 14 14:51:42 2021
Last Completed: Wed Jul 14 14:52:01 2021
Phase:          Succeeded
TEST SUITE:     chart-tests-test-ingress-v1beta1
Last Started:   Wed Jul 14 14:52:01 2021
Last Completed: Wed Jul 14 14:52:02 2021
Phase:          Succeeded
```

:partying_face:

File: charts/kong/templates/tests/test-ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        backend:
+          serviceName: "{{ .Release.Name }}-httpbin"
+          servicePort: 80
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ .Release.Name }}-test-ingress"
+  annotations:
+    "helm.sh/hook": test-success
```
Comment (rainest) [inline charts/kong/templates/tests/test-ingress.yaml]:
```suggestion
    "helm.sh/hook": test
```

File: charts/kong/templates/tests/test-ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      command:
+        - curl
+        - "http://{{ .Release.Name }}-kong-proxy.{{ .Release.Namespace }}.svc.cluster.local/httpbin"
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ .Release.Name }}-test-ingress-v1beta1"
+  annotations:
+    "helm.sh/hook": test-success
```
Comment (rainest) [inline charts/kong/templates/tests/test-ingress.yaml]:
```suggestion
    "helm.sh/hook": test
```

**Review by rainest — APPROVED** [review]
Minor changes to match the Helm 3 standard rather than using the still-supported Helm 2 test annotation.

I have some reticence about using large scripts for action steps rather than small steps--as we saw with the initial pipefail confusing failure message, it can be difficult to determine where exactly something failed in a bash script because bash is, well, bash. Smaller steps help isolate those failures. However, that's something we can always refactor later.

**Pull request merged**